hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/producers/formats/v6/BlockRecordWriterV6.java

```diff
@@ -0,0 +1,479 @@
+        final Path recordDir = fileSystem.getPath(config.logDir());
+        nodeScopedRecordDir = recordDir.resolve("record" + nodeInfo.memo());
+        nodeScopedSidecarDir = nodeScopedRecordDir.resolve(config.sidecarDir());
+
+        // Create parent directories if needed for the record file itself
+        try {
+            Files.createDirectories(nodeScopedRecordDir);
+        } catch (final IOException e) {
+            logger.fatal("Could not create record directory {}", nodeScopedRecordDir, e);
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    // =================================================================================================================
+    // Implementation of methods in BlockRecordWriter
+
+    /** {@inheritDoc} */
+    @Override
+    public void init(
+            @NonNull final SemanticVersion hapiProtoVersion,
+            @NonNull final HashObject startRunningHash,
+            @NonNull Instant startConsensusTime,
+            long blockNumber) {
+
+        if (state != State.UNINITIALIZED) {
+            throw new IllegalStateException("Cannot initialize a BlockRecordWriterV6 twice");
+        }
+
+        this.startObjectRunningHash = requireNonNull(startRunningHash);
+        this.startConsensusTime = requireNonNull(startConsensusTime);
+        this.hapiProtoVersion = requireNonNull(hapiProtoVersion);
+        this.blockNumber = blockNumber;
+        if (blockNumber < 0) {
+            throw new IllegalArgumentException("Block number must be non-negative");
+        }
+
+        // Create the chain of streams. In general, closing the "outermost" stream should close all the others, the
+        // same as flushing should. However, the HashingOutputStream does not behave correctly in this case, so we
+        // need to close all these streams individually, which means we have to maintain references to them.
+        this.recordFilePath = getRecordFilePath(startConsensusTime);
+        try {
+            fileOutputStream = Files.newOutputStream(recordFilePath);
+            if (compressFiles) {
+                gzipOutputStream = new GZIPOutputStream(fileOutputStream);
+                hashingOutputStream = new HashingOutputStream(createWholeFileMessageDigest(), gzipOutputStream);
+            } else {
+                hashingOutputStream = new HashingOutputStream(createWholeFileMessageDigest(), fileOutputStream);
+            }
+            bufferedOutputStream = new BufferedOutputStream(hashingOutputStream);
+            outputStream = new WritableStreamingData(bufferedOutputStream);
+
+            // Write the header
+            writeHeader(hapiProtoVersion);
+
+            state = State.OPEN;
+        } catch (final IOException e) {
+            logger.warn("Error initializing record file {}", recordFilePath, e);
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    @SuppressWarnings("java:S125")
```

Comment:
It's tedious, but easier for readers, if these suppressed sonar warnings come with a comment - e.g., in this case `// sections of code should not be comment out` - helps you not have to look further at what is being passed

(remark for future consideration)

Reply:
Great idea (for future consideration) :-)